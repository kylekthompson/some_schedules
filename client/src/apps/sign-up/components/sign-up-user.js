import PropTypes from 'prop-types';
import React from 'react';
import SignUpUserForm from 'apps/sign-up/components/sign-up-user-form';
import styled from 'styled-components';
import { Card, Container, Text } from 'components/style';
import { Separator } from 'components/form';
import { authenticated } from 'components/authentication';

const StyledContainer = styled(Container)`
  align-items: center;
  padding-top: 100px;
`;

const StyledCard = styled(Card)`
  align-items: center;
`;

export function SignUpUser({ requestSignIn }) {
  return (
    <StyledContainer>
      <StyledCard>
        <Text size={20} weight="bold">Sign Up - User</Text>
        <Separator />
        <SignUpUserForm onSuccess={requestSignIn} />
      </StyledCard>
    </StyledContainer>
  );
}

SignUpUser.propTypes = {
  requestSignIn: PropTypes.func.isRequired,
  user: PropTypes.object,
};

export default authenticated(SignUpUser);
