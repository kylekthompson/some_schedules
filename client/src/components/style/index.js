export { default as Button } from 'components/style/button';
export { default as Card } from 'components/style/card';
export { default as Container } from 'components/style/container';
export { default as Link } from 'components/style/link';
export { default as Text } from 'components/style/text';
