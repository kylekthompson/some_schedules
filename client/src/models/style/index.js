export { default as colors } from 'models/style/colors';
export { default as fonts } from 'models/style/fonts';
export { default as shadows } from 'models/style/shadows';
