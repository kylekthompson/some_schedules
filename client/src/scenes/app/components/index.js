export { default as Container } from 'scenes/App/components/Container';
export { default as HeaderContainer } from 'scenes/App/components/HeaderContainer';
export { default as HeaderLinks } from 'scenes/App/components/HeaderLinks';
