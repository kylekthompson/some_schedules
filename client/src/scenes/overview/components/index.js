export { default as Container } from 'scenes/Overview/components/Container';
export { default as HeaderLinks } from 'scenes/Overview/components/HeaderLinks';
export { default as Routes } from 'scenes/Overview/components/Routes';
