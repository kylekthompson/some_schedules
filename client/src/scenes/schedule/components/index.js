export { default as Container } from 'scenes/Schedule/components/Container';
export { default as ContentContainer } from 'scenes/Schedule/components/ContentContainer';
export { default as SidebarContainer } from 'scenes/Schedule/components/SidebarContainer';
