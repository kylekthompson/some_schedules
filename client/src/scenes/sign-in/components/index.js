export { default as Container } from 'scenes/sign-in/components/container';
