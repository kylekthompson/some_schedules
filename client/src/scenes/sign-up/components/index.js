export { default as Container } from 'scenes/sign-up/components/container';
