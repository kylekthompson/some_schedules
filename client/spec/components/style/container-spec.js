import React from 'react';
import { Container } from 'src/components/style';
import { mount } from 'spec/support/mount';

describe('<Container />', () => {
  describe('without any props', () => {
    it('is a div with flex 1 and flex direction column', () => {
      const { container } = mount(<Container />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('when passed a flex', () => {
    it('has the specified flex', () => {
      const { container } = mount(<Container flex="none" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('when passed a flexDirection', () => {
    it('has the specified flexDirection', () => {
      const { container } = mount(<Container flexDirection="row" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('when passed children', () => {
    it('renders the children', () => {
      const { getByText } = mount(<Container>Words</Container>);
      getByText('Words');
    });
  });
});
