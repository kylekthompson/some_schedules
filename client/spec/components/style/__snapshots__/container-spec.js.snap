// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> when passed a flex has the specified flex 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
/>
`;

exports[`<Container /> when passed a flexDirection has the specified flexDirection 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  class="c0"
/>
`;

exports[`<Container /> without any props is a div with flex 1 and flex direction column 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
/>
`;
